tests/message-extensions/src/index.ts

```diff
@@ -0,0 +1,129 @@
+import { cardAttachment } from '@microsoft/teams.api';
+import { App } from '@microsoft/teams.apps';
+import { Card } from '@microsoft/teams.cards';
+import { ConsoleLogger } from '@microsoft/teams.common/logging';
+import { DevtoolsPlugin } from '@microsoft/teams.dev';
+import {
+  createCard,
+  createConversationMembersCard,
+  createDummyCards,
+  createLinkUnfurlCard,
+  createMessageDetailsCard,
+} from './card';
+
+const app = new App({
+  logger: new ConsoleLogger('@tests/message-extensions', { level: 'debug' }),
+  plugins: [new DevtoolsPlugin()],
+});
+
+app.on('message', async ({ send, activity }) => {
+  await send({ type: 'typing' });
+  await send(`you said "${activity.text}"`);
+});
+
+// :snippet-start: message-ext-query-link
+app.on('message.ext.query-link', async ({ activity }) => {
+  const { url } = activity.value;
+
+  if (!url) {
+    return {
+      status: 400,
+      body: {},
+    };
+  }
+
+  const { card, thumbnail } = createLinkUnfurlCard(url);
+  const attachment = {
+    ...cardAttachment('adaptive', card),
+    preview: cardAttachment('thumbnail', thumbnail),
+  };
+
+  return {
+    composeExtension: {
+      type: 'result',
+      attachmentLayout: 'list',
+      attachments: [attachment],
+    },
+  };
+});
+// :snippet-end: message-ext-query-link
+// :snippet-start: message-ext-submit
+app.on('message.ext.submit', async ({ send, activity }) => {
+  const { commandId } = activity.value;
+  let card: Card;
+
+  if (commandId === 'createCard') {
+    // activity.value.commandContext == "compose"
+    card = createCard(activity.value.data);
+  } else if (commandId === 'getMessageDetails' && activity.value.messagePayload) {
+    // activity.value.commandContext == "message"
```

Comment:
Delete?

Reply:
i think it's a good idea to show what  the command context is, to reinforce what folks see in the manifest.

<img width="688" alt="Screenshot 2025-04-30 at 6 14 36 AM" src="https://github.com/user-attachments/assets/9bde9fa4-3382-4d78-bf4e-7d33c5ce2c0b" />

see "context"